adding typescript and some php (#1117)
temporalio/documentation: docs/typescript/how-to-develop-a-workflow-definition-in-typescript.md

---
id: how-to-develop-a-workflow-definition-in-typescript

> title: How to develop a Workflow definition in Typescript
> sidebar_label: Develop a Workflow definition
> description: Develop a Workflow definition

Reviewer: ```suggestion
title: How to develop a Workflow Definition in TypeScript
sidebar_label: Develop a Workflow Definition
description: Develop a Workflow Definition
```

tags:
  - developer-guide
  - sdk
  - typescript
---

A Workflow Function has two parts:

> - The function name is known as the Workflow Type.
> - The function implementation code (body) is known as the Workflow Definition.

Reviewer: ```suggestion
- The function name is the [Workflow Type](/docs/concepts/what-is-a-workflow-type/).
- The function implementation is the [Workflow Definition](/docs/concepts/what-is-a-workflow-definition).
```

"function implementation code" feels awk to me, and I'd say "body" is inaccurate, since I'd include the signature in the Definition.

> Each Workflow Definition is bundled with any third party dependencies, and registered by Workflow Type in a Worker. A Workflow function becomes a Workflow Execution (instance) only when started from a Workflow Client using its Workflow Type.

Reviewer: ```suggestion
Workflow Functions are bundled with their dependencies and registered by name in a Worker. A Workflow Function becomes a [Workflow Execution](/docs/concepts/what-is-a-workflow-execution) when it's started from a Workflow Client.
```

> Workflow Definitions are _just functions_, which can store state, and orchestrate Activity Functions.
> The following code snippet uses `proxyActivities` to create functions to schedule a `greet` Activity in the system to say hello.

Reviewer: ```suggestion
Workflow Functions are _just functions_, which can store state, and orchestrate Activity Functions.
The following code snippet uses `proxyActivities` to schedule a `greet` Activity in the system to say hello.
```

Reviewer: `proxyActivities` is missing from the snippet

> A Workflow Function can have multiple parameters, however, Temporal encourages you to use a single object parameter.

Reviewer: ```suggestion
A Workflow Function can have multiple parameters; however, we recommend using a single object parameter.
```

"Temporal encourages" makes me think something about the system encourages it, and makes me wonder what part of the system/how.

Reviewer: Move to below snippet?

```typescript
type ExampleArgs = {
  name: string;
};

export async function example(
  args: ExampleArgs
): Promise<{ greeting: string }> {
  const greeting = await greet(args.name);
  return { greeting };
}
```

> This Workflow Function takes the name of the parameters and assigns it to the variable name. Then it calls the function and pass the name of the variable as the argument. The function returns a promise with the value of `greeting`.

Reviewer: Purpose? Inclined to remove

Reviewer: In this case, feels to me like reading a description of what the code does is significantly harder to understand than reading the code.
